presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/QueryBuilder.java

```diff
@@ -173,9 +174,14 @@ private static Domain pushDownDomain(JdbcClient client, ConnectorSession session
 
     private List<String> toConjuncts(JdbcClient client, ConnectorSession session, List<JdbcColumnHandle> columns, TupleDomain<ColumnHandle> tupleDomain, List<TypeAndValue> accumulator)
     {
+        if (!tupleDomain.getDomains().isPresent()) {
```

Comment:
Why wasn't this needed before? Did your change uncover a pre-existing bug?

Reply:
This doesn't look correct. `!tupleDomain.getDomains().isPresent()` looks equivalent to `tupleDomain.isNone()`
- In the case of `isNone` domain, this code should return always false condition (`QueryBuilder#ALWAYS_FALSE`), here you're returning no conditions
- i guess this wasn't found before because when domain is `isNone`, the table scan should generally be replaced with empty values
   - why isn't this the case now?

Reply:
This looks like a bug in the engine. This test query
```sql
SELECT orderkey FROM orders WHERE orderkey IN (1.5, 2.3)
```
produces a plan with this:
```
- ScanFilter[table = jdbc:tpch.orders:TEST1496816030679104:TPCH:ORDERS, grouped = false, filterPredicate = false] => [orderkey:bigint]
```
The table scan should be eliminated, but it's not, so we fail when trying to process the constraint. But the fix here is incorrect, it should be false.

Reply:
I fixed the code here:
```java
if (tupleDomain.isNone()) {
    return ImmutableList.of(ALWAYS_FALSE);
}
```
@martint is looking into why the optimizer is not eliminating this

Reply:
https://github.com/prestosql/presto/pull/617